nvi-commons/src/main/java/no/sikt/nva/nvi/common/service/model/Candidate.java

```diff
@@ -305,13 +305,54 @@ private static Candidate updateCandidate(UpsertCandidateRequest request, Candida
         validateCandidate(request);
         var existingCandidateDao = repository.findByPublicationId(request.publicationId())
                                        .orElseThrow(CandidateNotFoundException::new);
-        if (shouldResetCandidate(request, existingCandidateDao) || isNotApplicable(existingCandidateDao)) {
+        if (shouldResetCandidate(request, existingCandidateDao)) {
             return resetCandidate(request, repository, periodRepository, existingCandidateDao);
+        }
+        if (isNotApplicable(existingCandidateDao)) {
+            return updateToNotApplicable(request, repository, periodRepository, existingCandidateDao);
         } else {
```

Comment:
This conditional has to be separated.
When resetting -> keeping periodYear.
When no longer applicable -> no periodYear to return